stdlib/CMakeLists.txt

```diff
@@ -18,6 +18,18 @@ if(SWIFT_BUILD_STATIC_STDLIB)
   list(APPEND SWIFT_STDLIB_LIBRARY_BUILD_TYPES STATIC)
 endif()
 
+# Also make a list of "stdlib extras".
+# This will determine if StdlibUnitTest, Platform/*,
+# SwiftPrivate, etc will build. 
+# This allows this logic to work with overlay only builds.
+set(SWIFT_STDLIB_EXTRAS_BUILD_TYPES)
+if(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY OR SWIFT_BUILD_DYNAMIC_STDLIB)
```

Comment:
This isn't quite going to work; not everything in the "stdlib extras" can build without the overlays.